### PR TITLE
[dynamo] support `cls.__flags__`

### PR DESCRIPTION
**Pull request by XuehaiPan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #133972
* __->__ #133970
* #133969

cc @rec @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @xmfan

### CODE REVIEW

**Review by jansel — CHANGES_REQUESTED** [review]
test failures?

File: torch/_dynamo/variables/builtin.py [hunk 1]
```diff
@@ -1719,6 +1719,8 @@ def call_getattr(
                         if source is not None:
                             return VariableBuilder(tx, source)(base)
                         return SourcelessBuilder.create(tx, base)
+                    if name == "__flags__":
+                        return ConstantVariable.create(value.__flags__)
```
Comment (jansel) [inline torch/_dynamo/variables/builtin.py:1723]: is this guaranteed to be constant?  do we need to guard on it?

Comment (XuehaiPan) [reply torch/_dynamo/variables/builtin.py:1731]: It is a readonly attribute:

```python
In [1]: class Foo:
   ...:     pass
   ...:     

In [2]: Foo.__flags__
Out[2]: 22040

In [3]: Foo.__flags__ = 1
AttributeError: readonly attribute
```

Comment (Skylion007) [reply torch/_dynamo/variables/builtin.py:1731]: Can we assert it's a read only attribute somehow?

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**